examples/snapshot-metadata-verifier/main.go

```diff
@@ -0,0 +1,177 @@
+	stringFlag(&args.TargetDevicePath, "target-device-path", "tgt", "", "The target device to use for verification.")
+
+	if home := homedir.HomeDir(); home != "" {
+		flag.StringVar(&kubeConfig, "kubeconfig", filepath.Join(home, ".kube", "config"), "Path to the kubeconfig file.")
+	} else {
+		flag.StringVar(&kubeConfig, "kubeconfig", "", "Path to the kubeconfig file.")
+	}
+
+	flag.StringVar(&args.SAName, "service-account", "", "ServiceAccount used to create a security token. If unspecified the ServiceAccount of the Pod in which the command is invoked will be used.")
+	flag.StringVar(&args.SANamespace, "service-account-namespace", "", "Namespace of the ServiceAccount used to create a security token. If unspecified the Namespace of the Pod in which the command is invoked will be used.")
+
+	flag.Int64Var(&args.TokenExpirySecs, "token-expiry", 600, "Expiry time in seconds for the security token.")
+	flag.Int64Var(&args.StartingOffset, "starting-offset", 0, "The starting byte offset.")
+
+	var maxResults int
+	flag.IntVar(&maxResults, "max-results", 0, "The maximum results per record.")
+
+	var showHelp bool
+	flag.BoolVar(&showHelp, "h", false, "Show the full usage message.")
+
+	progName := filepath.Base(os.Args[0])
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "\n"+shortUsageFmt, progName)
+	}
+
+	if len(os.Args) > 1 {
+		flag.Parse()
+		args.MaxResults = int32(maxResults)
+	} else {
+		fmt.Fprintf(os.Stderr, "Missing required arguments\n")
+		fmt.Fprintf(os.Stderr, shortUsageFmt, progName)
+		os.Exit(1)
+	}
+
+	if showHelp {
+		fmt.Fprintf(os.Stderr, usageFmt, progName)
+		flag.PrintDefaults()
+		os.Exit(0)
+	}
+}
+
+func main() {
+	// parse flags and set the output emitter
+	parseFlags()
+
+	// get the K8s config from either kubeConfig, in-cluster or default
+	config, err := buildConfig(kubeConfig)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error loading kubeconfig %s: %v\n", kubeConfig, err)
+		os.Exit(1)
+	}
+
+	clients, err := iterator.BuildClients(config)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error creating clients: %v\n", err)
+		os.Exit(1)
+	}
+
+	args.Clients = clients
+
+	sourceDevice, err := os.Open(args.SourceDevicePath)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to open source device %s: %q", args.SourceDevicePath, err)
+		os.Exit(1)
+	}
+	defer sourceDevice.Close()
+
+	targetDevice, err := os.OpenFile(args.TargetDevicePath, os.O_RDWR, 0644)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to open target device %s: %q", args.TargetDevicePath, err)
+		os.Exit(1)
+	}
+	defer targetDevice.Close()
+
+	args.Emitter = &verifier.VerifierEmitter{
+		SourceDevice: sourceDevice,
+		TargetDevice: targetDevice,
+	}
+
+	ctx, stopFn := signal.NotifyContext(context.Background(), os.Interrupt)
+	defer stopFn()
+
+	if err := verifier.VerifySnapshotMetadata(ctx, args); err != nil {
+		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+		os.Exit(1)
```

Comment:
I'm not sure why you feel the need embed the iterator in the `verifier` package! Would the following invocation not be the same? (Assuming that `New` is the exposed constructor in the `iterator` package).
```
if err := iterator.New(args).Run(ctx); err != nil {
...
```
i.e. the `verifier` package has supplied an `Emitter` that is capable of verifying source/target while the traversal is done by the `iterator` package.